src/components/Buttons/ClipboardButton.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// TODO: Extract to common shared UI library
+
+import { Check, Clipboard } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import ReactTooltip from 'react-tooltip'
+import styled from 'styled-components'
+
+import { useGlobalContext } from '../../contexts/global'
+
+const ClipboardButton = ({ textToCopy }: { textToCopy: string }) => {
+  const [hasBeenCopied, setHasBeenCopied] = useState(false)
+  const { setSnackbarMessage } = useGlobalContext()
+
+  const handleClick = () => {
+    navigator.clipboard
+      .writeText(textToCopy)
+      .catch((e) => {
+        throw e
+      })
+      .then(() => {
+        setHasBeenCopied(true)
+      })
+  }
+
+  useEffect(() => {
+    // Reset icon after copy
+    if (hasBeenCopied) {
+      ReactTooltip.rebuild()
+      setSnackbarMessage({ text: 'Copied to clipboard!', type: 'info' })
+
+      setInterval(() => {
+        setHasBeenCopied(false)
+      }, 3000)
+    }
+  }, [hasBeenCopied, setSnackbarMessage])
+
+  if (!hasBeenCopied) {
+    return <StyledClipboardIcon size={15} data-tip={'Copy to clipboard'} onClick={handleClick} />
+  } else {
+    return <StyledCheckIcon size={15} />
+  }
+}
+
+const StyledClipboardIcon = styled(Clipboard)`
+  margin-left: 10px;
+  cursor: pointer;
+  color: ${({ theme }) => theme.font.secondary};
+`
+
+const StyledCheckIcon = styled(Check)`
+  margin-left: 10px;
+  color: ${({ theme }) => theme.font.primary};
+`
+
+export default ClipboardButton
```

Comment:
Suggestion for this one, related to my other comment below. Curious to hear what you think about this way of wrapping the style + exporting + reusing shared style. 

```ts
import { Check, Clipboard } from 'lucide-react'
import { useEffect, useState } from 'react'
import ReactTooltip from 'react-tooltip'
import styled, { css } from 'styled-components'

import { useGlobalContext } from '../../contexts/global'

const ClipboardButton = ({ textToCopy, className }: { textToCopy: string; className?: string }) => {
  const [hasBeenCopied, setHasBeenCopied] = useState(false)
  const { setSnackbarMessage } = useGlobalContext()

  const handleClick = () => {
    navigator.clipboard
      .writeText(textToCopy)
      .catch((e) => {
        throw e
      })
      .then(() => {
        setHasBeenCopied(true)
      })
  }

  useEffect(() => {
    // Reset icon after copy
    if (hasBeenCopied) {
      ReactTooltip.rebuild()
      setSnackbarMessage({ text: 'Copied to clipboard!', type: 'info' })

      setInterval(() => {
        setHasBeenCopied(false)
      }, 3000)
    }
  }, [hasBeenCopied, setSnackbarMessage])

  return !hasBeenCopied ? (
    <StyledClipboardIcon className={className} size={15} data-tip={'Copy to clipboard'} onClick={handleClick} />
  ) : (
    <StyledCheckIcon className={className} size={15} />
  )
}

const BaseIconStyle = css`
  margin-left: 10px;
  color: ${({ theme }) => theme.font.secondary};
`

const StyledClipboardIcon = styled(Clipboard)`
  ${BaseIconStyle}
  cursor: pointer;
`

const StyledCheckIcon = styled(Check)`
  ${BaseIconStyle}
`

export default styled(ClipboardButton)``

```

Reply:
Nice catch, thanks for raising this topic!

First of all, the `StyledCheckIcon` has `color: ${({ theme }) => theme.font.primary};`, not `color: ${({ theme }) => theme.font.secondary};`, so the only common property is `margin-left: 10px;`.

To be honest, I don't rly like the approach of "base-styles" like this. The reason is that to understand the styles of the component I need to maintain things in my memory (base style) to construct the complete styles, because of the "flat" hierarchy of styles. Maybe it's because my brain is more used in the "traditional" nested SCSS patterns. I would much more prefer one of the 2 following options:

## Option 2
Wrapping React icon components in `styled()` so that we can then keep all styles in one block, similar to how we do with traditional SCSS with classes (`&.classname`) ([docs](https://styled-components.com/docs/advanced#caveat)):

```tsx
import { Check, Clipboard } from 'lucide-react'
...
const ClipboardButton = ({ textToCopy, className }: { textToCopy: string; className?: string }) => {
  ...
  if (!hasBeenCopied) {
    return <ClipboardIcon className={className} />
  } else {
    return <CheckIcon className={className} />
  }
}

const ClipboardIcon = styled(Clipboard)``
const CheckIcon = styled(Check)``

export default styled(ClipboardButton)`
  margin-left: 10px;

  &${ClipboardIcon} {
    cursor: pointer;
    color: ${({ theme }) => theme.font.secondary};
  }

  &${CheckIcon} {
    color: ${({ theme }) => theme.font.primary};
  }
`
```

## Option 3
Styled-components doesn't forbid us from using CSS classes:

```tsx
import { Check, Clipboard } from 'lucide-react'
...
const ClipboardButton = ({ textToCopy, className }: { textToCopy: string; className?: string }) => {
  ...
  if (!hasBeenCopied) {
    return (
      <Clipboard className={`${className} clipboard`} />
    )
  } else {
    return <Check className={`${className} check`} />
  }
}

export default styled(ClipboardButton)`
  margin-left: 10px;

  &.clipboard {
    cursor: pointer;
    color: ${({ theme }) => theme.font.secondary};
  }

  &.check {
    color: ${({ theme }) => theme.font.primary};
  }
`
```

My order of preference is:
- Option 3: For me this is the cleanest solution. It does not require unnecessary lines of code to wrap React components with `styled()`, while using plain old vanilla CSS/SCSS features. It also looks visually nice IMO :) Moreover, it only needs 11 lines of code (the other 2 options require 15).
- Option 2
- Option 1: The one you proposed.

Reply:
Interesting discussion :) 
I'm generally against defining static classnames as we're using styled-components to avoid just that. 
BUT in this case - and as we're anyway setting the `className` prop in both our child components - I believe that I prefer option 3 as well, less convoluted.